tools/path-encoding-verify.py

```
@@ -0,0 +1,116 @@
+VECTORS = ROOT / "test-vectors" / "path-encoding.json"
+
+# Reference form: RFC 3986 unreserved characters and uppercase %HH escapes only.
+SEGMENT = re.compile(r"^(?:[A-Za-z0-9._~-]|%[0-9A-F]{2})+$")
+# Encoder-variance form: additionally the five sub-delims encodeURIComponent leaves raw
+# (legal pchar in a path segment, decode to themselves — spec rule 4).
+ALT_SEGMENT = re.compile(r"^(?:[A-Za-z0-9._~!*'()-]|%[0-9A-F]{2})+$")
+# WHATWG URL Standard § 4.1: single-/double-dot path segments, ASCII case-insensitive.
+WHATWG_DOT_SEGMENTS = {".", "%2e", "..", ".%2e", "%2e.", "%2e%2e"}
+
+
+def check(condition: bool, name: str, detail: str) -> None:
```

Ruff (0.16.3)

[warning] Boolean-typed positional argument in function definition

(FBT001)